app/favorite/internal/repository/db/dao/favortie_detail.go

```diff
@@ -34,26 +35,42 @@ func (dao *FavoriteDetailDao) CreateFavoriteDetail(req *favoritePb.FavoriteDetai
 		Favorite: f,
 	}
 	err = dao.DB.Model(&model.FavoriteDetail{}).Create(&fd).Error
-
+	if err != nil {
+		return errors.Wrapf(err, "failed to create favoriteDetail，userID = %v,urlId = %v", req.UserId, req.UrlId)
+	}
 	return
 }
 
 func (dao *FavoriteDetailDao) ListFavoriteDetail(req *favoritePb.FavoriteDetailListReq) (r []*model.Favorite, err error) {
 	var f []*model.Favorite
-	dao.DB.Where("user_id = ?", req.UserId).Find(&f)
+	err = dao.DB.Where("user_id = ?", req.UserId).Find(&f).Error
+	if err != nil {
+		return r, errors.Wrapf(err, "failed to query favorite, userId = %v ", req.UserId)
+	}
 	for _, v := range f {
-		_ = dao.DB.Model(&v).Association("FavoriteDetail").Find(&v.FavoriteDetail)
+		err = dao.DB.Model(&v).Association("FavoriteDetail").Find(&v.FavoriteDetail)
```

Comment:
the variable `err` will reassign in the range. the root cause is code quality, remark as todo, maybe later i will optimum.